cabal-install/Distribution/Client/CmdConfigure.hs

```diff
@@ -95,7 +95,7 @@ configureAction flags@NixStyleFlags {..} _extraArgs globalFlags = do
     -- before overwriting
     exists <- doesFileExist "cabal.project.local"
     when exists $ do
```

Comment:
`when exists` let's search for the first one of `cabal.project.local~`, `cabal.project.local~0` `cabal.project.local~1` etc. until one doesn't.

Note: we need to change this to take into account `--project-file` as we on it. I wonder if there's an issue about that too.